format/FlightSql.proto

```diff
@@ -1418,7 +1442,17 @@ message ActionCreatePreparedStatementRequest {
 }
 
 /*
- * Wrap the result of a "GetPreparedStatement" action.
+ * Request message for the "CreatePreparedSubstraitPlan" action on a Flight SQL enabled backend.
+ */
+message ActionCreatePreparedSubstraitPlan {
+  option (experimental) = true;
+
+  // The serialized substrait.Plan to create a prepared statement for.
+  bytes plan = 1;
```

Comment:
I haven't been keeping up but is serialized plan sufficient, or are yaml plugins necessary?

Reply:
The root plan contains extension definitions, and I believe they're intended to be self-contained as a result, but I'll seek some clarification: https://github.com/substrait-io/substrait/blob/1080f06298d8e50abcd6acfaa6c425326a7e0579/proto/substrait/plan.proto#L24-L45

Reply:
I'd like to keep code dependencies on other Protobuf messages out because Windows/Protobuf has issues with those when they're in different DLLs (as they are with Flight/Flight SQL, and as they would be here). As seen below with the CancelQuery message I already ran into linking issues and I think they're insurmountable unless protoc itself is modified.

Reply:
On the wire, the actual encoding is the same either way.

Reply:
@lidavidm Fair enough, that makes sense.